### PR TITLE
Improve reported statistics

### PR DESCRIPTION
**Pull request by lexnv**
This PR adds several improvements that aim to report more insightful statistical data.

### Memory Buckets

Most reported systems by our telemetry endpoint have at least 64GiB of ram. To provide more insights into the RAM of the nodes, an extra bucket is added that contains values in the range [64; 128) GiB of ram.

### Kernel Version

The linux kernel version is a string of the following format: `5.10.0-8-02354235`.

The propagated version from the backend now includes only the kernel version, major version and minor version.
The patch (` - 8 - `) and specific identified ( ` - 02354235`) are omitted for better grouping of the available versions.

### CPU Vendor

To provide more insights into the specific number of CPUs of the substrate nodes, another entry is added which groups CPUs by vendor (intel / amd).

### Package lock

Update the NPM package lock in response to updating vulnerable dependencies.

### With this PR
Deployment in staging:

Screenshot 2023-11-03 at 15 12 29
6-e6bb5330fa88)

### Next Steps

- [ ] Check if providing all information from the backend to the frontend is a viable option (ie letting the frontend do the parsing)

Closes https://github.com/paritytech/substrate-telemetry/issues/535.

cc @paritytech/subxt-team

### CODE REVIEW

File: backend/telemetry_core/src/state/chain_stats.rs [hunk 1]
```diff
@@ -96,6 +96,27 @@ fn bucket_memory(memory: u64) -> (u32, Option<u32>) {
...
+}
+
+#[test]
+fn test_kernel_version_number() {
+    assert_eq!(kernel_version_number(&"5.10.0-8-amd64".into()), "5.10.0");
+    assert_eq!(kernel_version_number(&"5.10.0".into()), "5.10.0");
+}
+
+fn cpu_vendor(cpu: &Box<str>) -> &str {
+    if cpu.contains("Intel") || cpu.contains("intel") {
```
Comment (niklasad1) [inline backend/telemetry_core/src/state/chain_stats.rs:114]: maybe https://doc.rust-lang.org/std/string/struct.String.html#method.eq_ignore_ascii_case to keep the code a little simpler?

File: backend/telemetry_core/src/state/chain_stats.rs [hunk 2]
```diff
@@ -96,6 +96,27 @@ fn bucket_memory(memory: u64) -> (u32, Option<u32>) {
         48,
         56,
         64,
+        128,
+    }
+}
+
+fn kernel_version_number(version: &Box<str>) -> &str {
+    &version[0..version.find('-').unwrap_or(version.len())]
```
Comment (niklasad1) [inline backend/telemetry_core/src/state/chain_stats.rs:104]: "1.20.0+c144a98" is also a valid semver version but probably a weird edge not sure if's worth the effort to strip that one as well

linux kernel is probably not using it anyway but yeah technically possible

Comment (niklasad1) [reply backend/telemetry_core/src/state/chain_stats.rs:104]: so you could just use semver::Version here :)

File: frontend/src/components/Stats/Stats.tsx [hunk 3]
```diff
@@ -110,6 +110,14 @@ function formatMemory(value: Range): string {
   return min + ' GB';
 }
 
+function formatLinuxKernel(value: string): string {
+  let index = value.indexOf('-');
```
Comment (niklasad1) [inline frontend/src/components/Stats/Stats.tsx:114]: probably fine but see my annoying comment in the rust code as well.

File: backend/telemetry_core/src/state/chain_stats.rs (hunk 2 above)
Comment (lexnv) [reply backend/telemetry_core/src/state/chain_stats.rs]: By `semver::Version` you mean using another crate for this?

For now I've manually parsed the kernel version to take into account the plus sign :D

**Review by niklasad1 — APPROVED** [review]

**Review by jsdw — APPROVED** [review]
Looks good to me.

I think there is a way to deploy a specific branch to staging to test it. Let's see if that's possible before merging this to master, just to see what it's like (which might not be very informative on rococo, but just in case it shows that we should to tweak any other buckets)

**Pull request merged**